Source/ee/VuBasicBlock.cpp

```diff
@@ -198,7 +244,51 @@ void CVuBasicBlock::CompileRange(CMipsJitter* jitter)
 			jitter->PushRel(offsetof(CMIPS, m_State.savedIntRegTemp));
 			jitter->PullRel(offsetof(CMIPS, m_State.nCOP2VI[integerBranchDelayInfo.regIndex]));
 		}
-
+		
+		if (address == m_begin)
+		{
+			if(IsConditionalBranch(opcodeLo))
+			{
+				/*
+					jit code needs to do something like follows:
+				 
+					if(branchLoOps.readI0 == m_State.savedNextBlockIntRegIdx || branchLoOps.readI1 == m_State.savedNextBlockIntRegIdx)
+					{
+						m_State.nCOP2VI[m_State.savedNextBlockIntRegIdx] = m_State.savedNextBlockIntRegValTemp;
+					}
+				 */
+				jitter->PushRel(offsetof(CMIPS, m_State.savedNextBlockIntRegIdx));
+				jitter->PushCst(ranchLoOps.readI0);
+				jitter->BeginIf(Jitter::CONDITION_EQ);
+				{
+					jitter->PushRel(offsetof(CMIPS, m_State.savedNextBlockIntRegIdx));
+					jitter->PushCst(ranchLoOps.readI1);
+					jitter->BeginIf(Jitter::CONDITION_EQ);
+					{
+						jitter->PushRel(offsetof(CMIPS, m_State.savedNextBlockIntRegValTemp));
+						
+						// We want to do something like as follows ... except we can't reference the m_sate value here
+						// because we only know it at run time.
+						
+						//jitter->PullRel(offsetof(CMIPS, m_State.nCOP2VI[m_State.savedNextBlockIntRegIdx]));
+						
+						jitter->PushCst(offsetof(CMIPS, m_State.nCOP2VI[0]));
+						
+						jitter->PushRel(offsetof(CMIPS, m_State.savedNextBlockIntRegIdx));
+						jitter->PushCst(4);
+						jitter->Mult();
+						jitter->Add();
+						
+						// TODO: I think what we now have on the stack is the value and the offset where to store it.
```

Comment:
how do we write this?

Reply:
You can use this to push the value of `m_ctx.m_State.nCOP2VI[m_State.savedNextBlockIntRegIdx]` on the stack:
```
jitter->PushRelAddrRef(offsetof(CMIPS, m_State.nCOP2VI));
jitter->PushRel(offsetof(CMIPS, m_State.savedNextBlockIntRegIdx));
jitter->LoadFromRefIdx();
```

And you can use this to store a value from the stack to `m_ctx.m_State.nCOP2VI[m_State.savedNextBlockIntRegIdx]`:
```
jitter->PushRelAddrRef(offsetof(CMIPS, m_State.nCOP2VI));
jitter->PushRel(offsetof(CMIPS, m_State.savedNextBlockIntRegIdx));
jitter->PushIdx(2) //Get the value that was on the top of the stack before pushing those two things above
jitter->StoreAtRefIdx();
```